sdk/metric/internal/exemplar/rand_test.go

```diff
@@ -0,0 +1,59 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package exemplar
+
+import (
+	"context"
+	"math"
+	"sort"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestFixedSize(t *testing.T) {
+	t.Run("Int64", ReservoirTest[int64](func(n int) (Reservoir[int64], int) {
+		return FixedSize[int64](n), n
+	}))
+
+	t.Run("Float64", ReservoirTest[float64](func(n int) (Reservoir[float64], int) {
+		return FixedSize[float64](n), n
+	}))
+}
+
+func TestFixedSizeSamplingCorrectness(t *testing.T) {
+	intensity := 0.1
+	sampleSize := 1000
+
+	data := make([]float64, sampleSize*1000)
+	for i := range data {
+		data[i] = (-1.0 / intensity) * math.Log(random())
+	}
+	// Sort to avoid position bias.
+	sort.Float64s(data)
+
+	r := FixedSize[float64](sampleSize)
+	for _, value := range data {
+		r.Offer(context.Background(), staticTime, value, nil)
+	}
+
+	var sum float64
+	for _, m := range r.(*randRes[float64]).store {
+		sum += m.Value
+	}
+	mean := sum / float64(sampleSize)
+
+	assert.InDelta(t, 1/mean, intensity, 0.01)
```

Comment:
This test is flaky

```
$ go test -run=TestFixedSizeSamplingCorrectness -count=1000
--- FAIL: TestFixedSizeSamplingCorrectness (0.18s)
    rand_test.go:58: 
                Error Trace:    /home/rpajak/repos/opentelemetry-go/sdk/metric/internal/exemplar/rand_test.go:58
                Error:          Max difference between 0.11328473723564143 and 0.1 allowed is 0.01, but difference was 0.013284737235641428
                Test:           TestFixedSizeSamplingCorrectness
```

I hope that changing `sampleSize` to `10000` will be enough to make it stable. We can also consider making the delta bigger.

Reply:
This is looking for a delta of `0.01` which is 10*sigma. I'll relax the delta to be `0.02` (i.e. 5*sigma) which should still be a high enough confidence interval without the noise.

As for scaling the sample size, given it scales the standard deviation of the mean inverse to its square root I think we will be better served lowering the confidence interval instead (i.e. its already ~0.03*sigma).

Reply:
And it will also make the test execution faster 😉